### DISCUSSION TITLE
Prepublish script

### DISCUSSION THREAD

**Discussion by aklinker1** [jsr-io/jsr#854] [Q&A] [answered]
I'm writing a simple job queue library and want to include an HTML file that a server could use to display a management UI (showing in progress jobs, failed jobs, stats over time, etc). 

<details>
<summary>Think Sidekiq's UI.</summary>

![Sidekiq UI](https://github.com/sidekiq/sidekiq/raw/main/examples/web-ui.png)

</details>

I know I can build the HTML file with a separate build step, then include the output in the uploaded package to JSR, but is there a `prepublish` script I could use, [similar to NPM's `prepublishOnly` or `prepack`](https://docs.npmjs.com/cli/v7/using-npm/scripts#life-cycle-scripts), that guarentees I don't forget to run the build step before publishing?

I can always just use github actions and not publishing locally. Just curious if there's already a way to do something like this.

**Comment by BlackAsLight** [comment, marked as answer]
I don't think there is a way to do this currently. The recommended way to publish is via github actions where you can put all the prerequisites you want.

**Reply by aklinker1** [reply]
Yeah, that's what I figured. 

Also, I just realized I could create a publish task and use `deno task publish` instead of `deno publish`.
